src/main/java/com/getindata/connectors/http/sink/HttpSinkWriter.java

```diff
@@ -0,0 +1,118 @@
+package com.getindata.connectors.http.sink;
+
+import lombok.EqualsAndHashCode;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.flink.api.connector.sink2.Sink;
+import org.apache.flink.connector.base.sink.writer.AsyncSinkWriter;
+import org.apache.flink.connector.base.sink.writer.BufferedRequestState;
+import org.apache.flink.connector.base.sink.writer.ElementConverter;
+
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+@Slf4j
+public class HttpSinkWriter<InputT> extends AsyncSinkWriter<InputT, HttpSinkRequestEntry> {
+  private final String endpointUrl;
+  private final HttpClient httpClient;
+
+  public HttpSinkWriter(
+      ElementConverter<InputT, HttpSinkRequestEntry> elementConverter, Sink.InitContext context, int maxBatchSize,
+      int maxInFlightRequests, int maxBufferedRequests, long maxBatchSizeInBytes, long maxTimeInBufferMS,
+      long maxRecordSizeInBytes, String endpointUrl
+  ) {
+    this(elementConverter, context, maxBatchSize, maxInFlightRequests, maxBufferedRequests, maxBatchSizeInBytes,
+         maxTimeInBufferMS, maxRecordSizeInBytes, endpointUrl, Collections.emptyList()
+    );
+  }
+
+  public HttpSinkWriter(
+      ElementConverter<InputT, HttpSinkRequestEntry> elementConverter, Sink.InitContext context, int maxBatchSize,
+      int maxInFlightRequests, int maxBufferedRequests, long maxBatchSizeInBytes, long maxTimeInBufferMS,
+      long maxRecordSizeInBytes, String endpointUrl,
+      Collection<BufferedRequestState<HttpSinkRequestEntry>> bufferedRequestStates
+  ) {
+    super(elementConverter, context, maxBatchSize, maxInFlightRequests, maxBufferedRequests, maxBatchSizeInBytes,
+          maxTimeInBufferMS, maxRecordSizeInBytes, bufferedRequestStates
+    );
+    this.endpointUrl = endpointUrl;
+    this.httpClient = HttpClient.newBuilder().followRedirects(HttpClient.Redirect.NORMAL).build();
```

Comment:
Please use dependency injection here pattern here.

Your implementation of `HttpSinkWriter` is now coupled with `java.net.http.HttpClient` which is external library. What if we woudl liek to use different client implementation? Or we would like to allow end users to used their favorite ones?

In Table Source we used abstraction, and we inject "client".
please take a look at `PollingClient` and `PollingClientFactory` interfaces and how they were used.

Reply:
We also have to consider how we would like to pass configuration to the client. 
For example we would have to allow for configuring "Cross Origin" or any other headers that might be required. 

the first step for this would be to abstract out initialization of HTTP client.